lib/rubocop/cop/performance/chain_array_allocation.rb

```diff
@@ -47,8 +47,9 @@ class ChainArrayAllocation < Cop
         HAS_MUTATION_ALTERNATIVE = ':collect :compact :flatten :map :reject '\
                                    ':reverse :rotate :select :shuffle :sort '\
                                    ':uniq '.freeze
-        MSG = 'Use `%<method>s...%<second_method>s!` instead of `%<method>s' \
-              '...%<second_method>s`.'.freeze
+        MSG = 'Use unchained `%<method>s!` and `%<second_method>s!` '\
+              '(followed by `return array` if required) instead of chaining '\
+              '`%<method>s...%<second_method>s`.'.freeze
```

Comment:
```
Use unchained `map!` and `flatten!` (followed by `return array` if required) 
instead of chaining `map...flatten`.`
```

This is the message I came up with for now, let me know if you need a better one.